packages/core/src/ctype/CType.ts

```diff
@@ -71,21 +56,34 @@ export function getHashForSchema(
  * @param hash CType hash as hex string.
  * @returns Schema id uri.
  */
-export function getIdForCTypeHash(
-  hash: ICType['hash']
-): ICType['schema']['$id'] {
+export function getIdForCTypeHash(hash: CTypeHash): ICType['$id'] {
   return `kilt:ctype:${hash}`
 }
 
+/**
+ * Extracts the CType hash from a CType $id.
+ *
+ * @param id A CType id of the form 'kilt:ctype:0x[0-9a-f]'.
+ * @returns The CType hash as a zero-prefixed string of hex digits.
+ */
+export function getCTypeHashFromId(id: ICType['$id']): CTypeHash {
```

Comment:
Since this is likely to be used as `CType.getCTypeHashFromId()` I’d prefer to rename it to `getHashFromId`. Actually if we’re using `*ToChain()` functions, maybe this could be `idToHash()`? The other one could be `hashToId()` then.

Reply:
it's definitely shorter